SEACAS: header "fmt/ostream.h" not found in Nalu and Albany builds

## Bug Report
In file included from /ascldap/users/spdomin/gitHubWork/scratch_build/install/gcc10.3.0/Trilinos_nightly_release/include/Ioss_StructuredBlock.h:16,
                 from /ascldap/users/spdomin/gitHubWork/scratch_build/install/gcc10.3.0/Trilinos_nightly_release/include/Ioss_SubSystem.h:35,
                 from /home/spdomin/gitHubWork/nightlyBuildAndTest/NaluNightly/src/InputOutputRealm.C:24:
/ascldap/users/spdomin/gitHubWork/scratch_build/install/gcc10.3.0/Trilinos_nightly_release/include/Ioss_ZoneConnectivity.h:14:10: fatal error: fmt/ostream.h: No such file or directory
   14 | #include <fmt/ostream.h>

Good:
NaluCFD/Nalu SHA1: 8959168ab0f53a39178694457d276d4b887bcf1a
Trilinos/develop SHA1: e9b4687e7a95c0626abd0bc64da33e42470895fe

Bad:
NaluCFD/Nalu SHA1: 8959168ab0f53a39178694457d276d4b887bcf1a
Trilinos/develop SHA1: 7ab1eeca884dbfff6eedfd906f7affa10c773b44

Perhaps @alanw0 could advise.

best,

> Fyi, manually copying the header `ostream.h` into `INSTALL/include/fmt/ostream.h` reportedly fixes this error in the Albany build on Perlmutter.

This error is showing up now after manually copying the header. Not sure if it is related to this issue but I figured I'd point it out just in case.
```/usr/bin/ld: ../../../src/libalbanyLib.so: undefined reference to `Albany::getMpiCommFromTeuchosComm(Teuchos::RCP<Teuchos::Comm<int> const>&)'
collect2: error: ld returned 1 exit status
make[2]: *** [tests/unit/evaluators/CMakeFiles/scatterResidual_unit_tester.dir/build.make:227: tests/unit/evaluators/scatterResidual_unit_tester] Error 1
make[1]: *** [CMakeFiles/Makefile2:2125: tests/unit/evaluators/CMakeFiles/scatterResidual_unit_tester.dir/all] Error 2```

This error is back:

Bad: Trilinos/develop SHA1: 94b60d005ec42fe20eda3a869712ffab8548b271
Good: Trilinos/develop SHA1: c8548cf7bdc5a50daff9fdf93d493228a74a3973

In file included from /projects/netpub/nalu/scratch_build/install/gcc10.3.0/Trilinos_nightly_release/include/stk_io/IossBridge.hpp:41,
                 from /fgs/spdomin/nightly/Nalu/src/DataProbePostProcessing.C:33:
/projects/netpub/nalu/scratch_build/install/gcc10.3.0/Trilinos_nightly_release/include/Ioss_Field.h:12:10: fatal error: fmt/ostream.h: No such file or directory
   12 | #include <fmt/ostream.h>

Perhaps this hit Albany as well again? @ikalash

--- a/packages/seacas/libraries/ioss/src/Ioss_Field.h
+++ b/packages/seacas/libraries/ioss/src/Ioss_Field.h
@@ -8,9 +8,6 @@
 
 #include "Ioss_CodeTypes.h"
 #include <cstddef> // for size_t
-#if !defined BUILT_IN_SIERRA
-#include <fmt/ostream.h>
-#endif
 #include <stdint.h>
 #include <string> // for string
 #include <vector> // for vector
@@ -247,13 +244,3 @@ namespace Ioss {
   };
   IOSS_EXPORT std::ostream &operator<<(std::ostream &os, const Field &fld);
 } // namespace Ioss
-
-#if !defined BUILT_IN_SIERRA
-#if FMT_VERSION >= 90000
-namespace fmt {
-  template <> struct formatter<Ioss::Field> : ostream_formatter
-  {
-  };
-} // namespace fmt
-#endif
-#endif

--- a/packages/seacas/libraries/ioss/src/Ioss_Region.h
+++ b/packages/seacas/libraries/ioss/src/Ioss_Region.h
@@ -24,9 +24,6 @@
 #include "Ioss_Utils.h"
 #include "Ioss_VariableType.h"
 #include "ioss_export.h"
-#if !defined BUILT_IN_SIERRA
-#include <fmt/ostream.h>
-#endif
 #include <functional> // for less
 #include <iosfwd>     // for ostream
 #include <map>        // for map, map<>::value_compare
@@ -484,17 +481,9 @@ namespace Ioss {
 
         if (found && field.get_role() != role) {
           std::ostringstream errmsg;
-#if defined BUILT_IN_SIERRA
           errmsg << "ERROR: Field " << field.get_name() << " with role " << field.role_string()
                  << " on entity " << entity->name() << " does not match previously found role "
                  << Ioss::Field::role_string(role) << ".\n",
-#else
-          fmt::print(errmsg,
-                     "ERROR: Field {} with role {} on entity {} does not match previously found "
-                     "role {}.\n",
-                     field.get_name(), field.role_string(), entity->name(),
-                     Ioss::Field::role_string(role));
-#endif
               IOSS_ERROR(errmsg);
         }
 

--- a/packages/seacas/libraries/ioss/src/Ioss_StructuredBlock.h
+++ b/packages/seacas/libraries/ioss/src/Ioss_StructuredBlock.h
@@ -12,7 +12,6 @@
 #include "Ioss_NodeBlock.h"
 #include "Ioss_Property.h"
 #include "Ioss_ZoneConnectivity.h"
-#include <fmt/ostream.h>
 #include <array>
 #include <cassert>
 #include <iosfwd>
@@ -350,10 +349,3 @@ namespace Ioss {
   };
 } // namespace Ioss
 
-#if FMT_VERSION >= 90000
-namespace fmt {
-  template <> struct formatter<Ioss::BoundaryCondition> : ostream_formatter
-  {
-  };
-} // namespace fmt
-#endif

--- a/packages/seacas/libraries/ioss/src/Ioss_Utils.C
+++ b/packages/seacas/libraries/ioss/src/Ioss_Utils.C
@@ -1369,7 +1369,7 @@ void Ioss::Utils::info_fields(const Ioss::GroupingEntity *ige, Ioss::Field::Role
   for (const auto &field_name : fields) {
     if (detail) {
       const auto &field_ref = ige->get_fieldref(field_name);
-      fmt::print("{}{}", field_ref, suffix);
+      std::cout << field_ref << suffix;
     }
     else {
       const Ioss::VariableType *var_type   = ige->get_field(field_name).raw_storage();

--- a/packages/seacas/libraries/ioss/src/Ioss_ZoneConnectivity.h
+++ b/packages/seacas/libraries/ioss/src/Ioss_ZoneConnectivity.h
@@ -10,8 +10,6 @@
 #include <array>
 #include <cassert>
 #include <cmath>
-#include <fmt/core.h>
-#include <fmt/ostream.h>
 #include <iosfwd>
 #include <stdlib.h>
 #include <string>
@@ -142,11 +140,3 @@ namespace Ioss {
 
   IOSS_EXPORT std::ostream &operator<<(std::ostream &os, const ZoneConnectivity &zgc);
 } // namespace Ioss
-
-#if FMT_VERSION >= 90000
-namespace fmt {
-  template <> struct formatter<Ioss::ZoneConnectivity> : ostream_formatter
-  {
-  };
-} // namespace fmt
-#endif

--- a/packages/seacas/libraries/ioss/src/main/io_info.C
+++ b/packages/seacas/libraries/ioss/src/main/io_info.C
@@ -317,11 +317,7 @@ namespace {
       if (!sb->m_zoneConnectivity.empty()) {
         fmt::print("\tConnectivity with other blocks:\n");
         for (const auto &zgc : sb->m_zoneConnectivity) {
-#if defined __NVCC__
           std::cout << zgc << "\n";
-#else
-          fmt::print("{}\n", zgc);
-#endif
         }
       }
       if (!sb->m_boundaryConditions.empty()) {
@@ -335,11 +331,7 @@ namespace {
                    });
 
         for (const auto &bc : sb_bc) {
-#if defined __NVCC__
           std::cout << bc << "\n";
-#else
-          fmt::print("{}\n", bc);
-#endif
         }
       }
       if (interFace.compute_bbox()) {

